src/Hpack/Config.hs

```diff
@@ -566,12 +573,23 @@ mkPackage dir (CaptureUnknownFields unknownFields globalOptions@Section{sectionD
       where
         fromGithub = (++ "/issues") . sourceRepositoryUrl <$> github
 
-toLibrary :: FilePath -> String -> Section global -> Section LibrarySection -> IO (Section Library)
-toLibrary dir name globalOptions library = traverse fromLibrarySection sect
+-- | Expand the @c-sources@ globs in a 'Section' and return the modified
+-- 'Section' along with any warnings emitted.
+globSectionCSources :: FilePath -> Section a -> IO ([String], Section a)
+globSectionCSources dir sect = do
```

Comment:
I would remove the comment and rename the function to something like `expandCSources`